data-access-layer/iaas/AzureClient.js

```diff
@@ -27,6 +27,58 @@ class AzureClient extends BaseCloudClient {
       .value()
     );
     Promise.promisifyAll(this.computeClient.snapshots);
+    Promise.promisifyAll(this.computeClient.disks);
+  }
+
+  getSnapshot(snapshotId) {
+    return this.computeClient.snapshots.getAsync(this.settings.resource_group, snapshotId)
+      .then(snapshotResponse => _.pick(snapshotResponse, ['location', 'creationData', 'id']));
+  }
+
+  createDiskFromSnapshot(snapshotId, zones, opts = {}) {
+    const diskName = this.getRandomDiskId();
+    return Promise.try(() => this.getSnapshot(snapshotId))
+      .then(snapshotData => {
+        const options = {
+          zones: _.isArray(zones) ? zones : [zones],
+          location: snapshotData.location,
+          tags: _.assign({}, opts.tags || {}, {
+            createdBy: 'service-fabrik'
+          }),
+          sku: _.assign({}, opts.sku || {}, {
+            name: 'Premium_LRS',
+            tier: 'Premium'
+          }),
+          creationData: {
+            createOption: 'Copy',
+            sourceUri: snapshotData.id
+          }
+        };
+        return this.computeClient.disks.createOrUpdateAsync(this.settings.resource_group, diskName, options);
+      })
+      .then(diskResponse => ({
```

Comment:
I don't think we need to separately since the promise resolves only when the state is finalized. Still, I will check further in my individual tests and reply